src/main/java/de/javakaffee/kryoserializers/guava/ImmutableSetSerializer.java

```diff
@@ -0,0 +1,76 @@
+package de.javakaffee.kryoserializers.guava;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+
+/**
+ * A kryo {@link Serializer} for guava-libraries {@link ImmutableSet}.
+ */
+public class ImmutableSetSerializer extends Serializer<ImmutableSet<Object>> {
+
+    private static final boolean DOES_NOT_ACCEPT_NULL = false;
+    private static final boolean IMMUTABLE = true;
+
+    public ImmutableSetSerializer() {
+        super(DOES_NOT_ACCEPT_NULL, IMMUTABLE);
+    }
+
+    @Override
+    public void write(Kryo kryo, Output output, ImmutableSet<Object> object) {
+        output.writeInt(object.size(), true);
+        for (Object elm : object) {
+            kryo.writeClassAndObject(output, elm);
+        }
+    }
+
+    @Override
+    public ImmutableSet<Object> read(Kryo kryo, Input input, Class<ImmutableSet<Object>> type) {
+        final int size = input.readInt(true);
+        final Object[] set = new Object[size];
```

Comment:
ImmutableListSerializer came in via a pull request as well. 

Looking at the source, ImmutableSet builder would be a little bit more efficient for less than five elements (the array.clone as done by copyOf is saved), for more elements copyOf seems to get more efficient. So we can use the copyOf... I'll merge and make a release soon.